### PR TITLE
Use generic `Enum` methods where possible.

### PR DESCRIPTION
**Pull request by teo-tsirpanis**
This PR updates all non-test projects that exclusively target .NET 6 and above to use the generic `Enum` methods (#2364), reducing boxing and (perhaps?) making the code more AOT-friendly.

### CODE REVIEW

File: src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ToolboxItemFilterAttribute.cs [hunk 1]
```diff
@@ -104,6 +104,6 @@ public override bool Match([NotNullWhen(true)] object? obj)
                 && other.FilterString.Equals(FilterString);
         }
 
-        public override string ToString() => FilterString + "," + Enum.GetName(typeof(ToolboxItemFilterType), FilterType);
+        public override string ToString() => FilterString + "," + Enum.GetName<ToolboxItemFilterType(FilterType);
```
Comment (gfoidl) [inline src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ToolboxItemFilterAttribute.cs:107]:
```suggestion
        public override string ToString() => FilterString + "," + Enum.GetName<ToolboxItemFilterType>(FilterType);
```

Comment (teo-tsirpanis) [reply src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/ToolboxItemFilterAttribute.cs]: Thank you!

File: src/libraries/System.Private.Xml/src/System/Xml/Xsl/IlGen/OptimizerPatterns.cs [hunk 2]
```diff
@@ -54,7 +54,7 @@ internal enum OptimizerPatternArgument
     /// </summary>
     internal sealed class OptimizerPatterns : IQilAnnotation
     {
-        private static readonly int s_patternCount = Enum.GetValues(typeof(OptimizerPatternName)).Length;
+        private static readonly int s_patternCount = Enum.GetValues<OptimizerPatternName>().Length;
```
Comment (stephentoub) [inline src/libraries/System.Private.Xml/src/System/Xml/Xsl/IlGen/OptimizerPatterns.cs:57]: > reducing boxing

Note this doesn't current help avoid any boxing, but it's fine.

**Review by jeffhandley — APPROVED** [review]

**Review by stephentoub — APPROVED** [review]
Thanks.

**Comment by stephentoub** [comment]
@steveisok, the runtime-staging legs keep timing out on this PR.  Is there a known issue here?  Is it possible this PR is causing it?

**Comment by steveisok** [comment]
I'd update the branch to the latest main.  The helix queue being used is different.

**Comment by stephentoub** [comment]
> I'd update the branch to the latest main. The helix queue being used is different.

Shouldn't that have happened automatically when I closed and re-opened the PR?  Or does rebasing on the latest main no longer happen automatically?

Given that these are stale and everything else passed and there's nothing in the PR that should be mono-specific, I'm just going to merge.

**Comment by teo-tsirpanis** [comment]
That was why you closed and reopened it twice. 😅 Didn't know that was ever a thing.

**Comment by danmoseley** [comment]
@teo-tsirpanis it's easier than pushing a blank change -- I even do it on my phone! The other option is an `/azp` command, but that requires committer access plus it's harder to remember.

**Comment by teo-tsirpanis** [comment]
When Stephen said about "rebasing on the latest main" my first thought was that it would rebase my fork's branch and was confused, but turns out he meant it would run CI again with my changes incorporated to the latest main, and makes more sense.

**Comment by danmoseley** [comment]
Got it, yes, I believe each time that validation runs, it behind the scenes rebases on the latest HEAD before it builds.

**Pull request merged**